tutorials/haskell/hspec-golden-test-tutorial/tutorial.md

```diff
@@ -0,0 +1,506 @@
+       </body>"
+      but got: "
+       <body>
+               Goodbye Golden Testers.
+       </body>"
+
+
+Finished in 0.0013 seconds
+3 examples, 1 failure
+```
+
+Our HTML test failed but not big deal, we can edit our `golden` file and things will work just fine. Pretty simple right?
+
+Actually it's not, let's remember this is just a didactic example, in the real life golden files stores really large outputs and it would be a waste of time to update them by hand. Fortunately, `hspec-golden` provides us with `hgold` CLI tool that automatically updates golden files using. Let's install it.
+
+Using stack:
+
+```shell
+$stack install hspec-golden
+```
+
+using Cabal:
+
+```shell
+$cabal install hspec-golden
+```
+Once installed we can use the `--help` flag to see how the CLI works.
+
+According to [hspec-golden](http://hackage.haskell.org/package/hspec-golden), when `hgold` is used without flags it updates by default the `.golden/` directory. If we store our golden files in a different directory we should use the `--update` flag and specify the name of the directory as an argument.
+
+Okay, let's update our golden files. 
+
+```shell
+$ hgold
+Replacing golden with actual...
+  Replacing file: .golden/hello/golden with: .golden/hello/actual
+  Replacing file: .golden/html/golden with: .golden/html/actual
+Finish...
+```
+
+Hgold replaces the `golden` file (expected output) with the content of the `actual` file (SUT output) so it is necessary to run the test first (even if they fail) to update our tests.
+
+So now we are ready to test our updated golden files.
+
+```shell 
+Hello.HelloGolden
+  sayHi
+    returns Hello Golden Testers string
+      Golden and Actual output hasn't changed
+Hello.Hello
+  sayHi
+    shows a Hello Golden Testers string
+Html.HtmlGolden
+  renderHtml
+    Renders an Html5 file 
+      Golden and Actual output hasn't changed
+
+Finished in 0.0034 seconds
+3 examples, 0 failures
+```
+
+And we can see that everything works fine.
+
+#### JSON Golden Test
+
+Until now, we have managed components that returns strings only, but what if our return type is different like a `ByteString` or `Text`? For example, let's analize the `encodeCountries` function in the `Json.hs` module.
+
+```Haskell
+encodeCountries :: [Country] -> ByteString
+encodeCountries = encode
+```
+
+This function returns a ByteString, that means that if we try to assert the output with our `defaultGolden` function,
+
+``` Haskell
+  describe "encodeCountries" $ do
+   it "encodes a group of Countries into a JSON String " $
+    defaultGolden "json" (encodeCountries countries)
+```
+
+test won't even compile.
+
+```shell
+$stack test
+• Couldn't match type ‘B.ByteString’ with ‘[Char]’
+      Expected type: String
+        Actual type: B.ByteString
+```
+
+Conviniently, `hspec-golden` exports a `Golden` data type. With this tool we can configure our Golden test 
+specifying the functions `hspec-golden` should use for our test to work.
+
+```
```

Comment:
@Jagl257 I think you need to specify the type of this code (e.g. haskell).

Reply:
@CristhianMotoche yes Cris it a haskell type. Thank's